core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/output/ToCsvHandler.java

```diff
@@ -73,15 +74,18 @@ private LinkedHashMap<String, String> getPropertyHeadersFromSchema(final Store s
             }
         }
         for (final SchemaEdgeDefinition schemaEdgeDefinition : schema.getEdges().values()) {
-            for (final IdentifierType identifierType:schemaEdgeDefinition.getIdentifiers()) {
+            for (final IdentifierType identifierType : schemaEdgeDefinition.getIdentifiers()) {
                 if (identifierType.toString().equals(identifierType.DIRECTED.toString())) {
-                    String typeName = schemaEdgeDefinition.getIdentifierTypeName(identifierType);
+                    final String typeName = schemaEdgeDefinition.getIdentifierTypeName(identifierType);
                     propertyHeadersFromSchema.put(identifierType.toString(), schema.getType(typeName).getClazz().getSimpleName());
                 }
             }
-            for (final String propertyName:schemaEdgeDefinition.getProperties()) {
-                String typeName = schemaEdgeDefinition.getPropertyTypeName(propertyName);
-                propertyHeadersFromSchema.put(propertyName, schema.getType(typeName).getClazz().getSimpleName());
+            for (final String propertyName : schemaEdgeDefinition.getProperties()) {
+                final String typeName = schemaEdgeDefinition.getPropertyTypeName(propertyName);
+                final String javaType = schema.getType(typeName).getClazz().getSimpleName();
+                final String openCypherType = OpenCypherFormat.typeMappings.getOrDefault(javaType, javaType);
+
+                propertyHeadersFromSchema.put(propertyName, openCypherType);
```

Comment:
This change to use the `OpenCypherFormat.typeMappings` is causing tests to fail because they don't expect the OpenCypher `Int`.

The test failure is confusing because the asserts in there are also backwards, causing the error to reported that it expected `Int` but was `Integer`, when it is actually the other way around in the code.

I think it would be better if the mapping between Java and OpenCypher types was done in a separate for loop after these ones. Is there a reason why the identifier type doesn't need to be converted to a OpenCypher one?

Reply:
The type mapping is only relevant for properties as only property headings have types. I have sorted the broken tests now